src/ui/src/builder/panels/BuilderCodePanel.vue

```diff
@@ -280,6 +292,22 @@ async function handleSave() {
 		isDisabled.value = false;
 	}
 }
+
+function checkRenamePath(pathSegments: string[]): string | null {
+	if (pathSegments[0] === "") {
+		return "Path should not have a leading slash";
+	}
+
+	if (pathSegments[0] === ".wf") {
+		return ".wf is reserved for Writer Framework";
+	}
+
+	if (pathSegments.some((segment) => segment === "")) {
+		return "Invalid path";
+	}
+
+	return null;
```

Comment:
Optional, but I think you can shorten it

```suggestion
	if (pathSegments[0] === "") return "Path should not have a leading slash";
	if (pathSegments[0] === ".wf") return ".wf is reserved for Writer Framework";
	if (pathSegments.some((segment) => segment === "")) return "Invalid path";
```